### PR TITLE
[release-1.3] add protocolDetectionTimeout

### PR DESCRIPTION
**Pull request by yxue**

### CODE REVIEW

File: global.yaml [hunk 1]
```diff
@@ -111,6 +111,17 @@ global:
...
+    # Automatic protocol detection uses a set of heuristics to
+    # determine whether the connection is using TLS or not (on the
+    # server side), as well as the application protocol being used
+    # (e.g., http vs tcp). These heuristics rely on the client sending
+    # the first bits of data. For server first protocols like MySQL,
+    # MongoDB, etc., Envoy will timeout on the protocol detection after
+    # the specified period, defaulting to non mTLS plain TCP
+    # traffic. Set this field to tweak the period that Envoy will wait
+    # for the client to send the first bits of data. (MUST BE >=1ms)
+    protocolDetectionTimeout: 10s
```
Comment (howardjohn) [inline global.yaml]:
```suggestion
    protocolDetectionTimeout: 10s
```

Comment (yxue) [reply global.yaml]: Just for testing

Comment (yxue) [reply global.yaml]: Will revert

**Comment by yxue** [comment]
@rlenglet The fix works. I sent a PR to master branch to fix and will cherry-pick after being merged. https://github.com/istio/installer/pull/367

**Review by rlenglet — APPROVED** [review]

**Comment by rlenglet** [comment]
This PR keeps failing. I had to restart the following tests 5~6 times so far:
- `ci/circleci: integration-presubmit-isolated-namespaces-kind`
- `ci/circleci: integration-presubmit-one-namespace-kind`
- `ci/circleci: noauth`

This is not normal.

**Comment by rlenglet** [comment]
150ms seems too short for these tests to pass.

**Comment by howardjohn** [comment]
I don't think it's related to protocol sniffing... it's the namespace issue for citadel, didn't Sam cherry pick the pr?

**Comment by rlenglet** [comment]
Oh  yes I think we cherrypicked that.

**Pull request closed without merging**